tests/integration/suite/test_catalog.py

```diff
@@ -291,22 +292,26 @@ def test_invalid_catalog_charts(admin_mc, remove_resource):
     client = admin_mc.client
     name = random_str()
     url = "https://github.com/rancher/integration-test-charts"
-
     catalog = client.create_catalog(name=name,
                                     branch="broke-charts",
                                     url=url,
                                     )
     remove_resource(catalog)
     wait_for_template_to_be_created(client, catalog.id)
 
-    catalog = client.reload(catalog)
+    def get_errored_catalog(catalog):
+        while catalog.transitioning != "error":
```

Comment:
If catalog never goes to error this will never exit. This should refresh the catalog and return a bool/catalog based on the check of `catalog.transitioning`